.github/workflows/email.yml

```diff
@@ -24,8 +24,22 @@ jobs:
         env:
           COMMIT_URL: ${{github.event.pull_request._links.commits.href}}
           LOG: ${{github.event.pull_request.body}}
-        run: |
+        uses: nick-fields/retry@v2
+        with:
+          max_attempts: 4
+          retry_on_exit_code: 2
+          timeout_seconds: 60  
+          command: |
               curl --location --request GET $COMMIT_URL --header 'X-API-Key: ${{ secrets.GITHUB_TOKEN}}' -o commits.json
+              if [ $? -ne 0 ]; then
+                echo "curl command to get commits payload failed"
+                exit 2
+              fi  
+              jq -r '.[0].commit.author.name' commits.json  
+              if [ $? -ne 0 ]  ; then
+                echo "error in processing .jq command"
+                exit 2
+              fi       
```

Comment:
This will retry the step  get commits payload in case of any error with parsing commit.json using jquerry command.
We are running into issues where the curl command above timesout and the commit.json is empty causing the following jquerry commands to fail.
This causes the from addess(required field) in email step to fail.